dips/dip-12.md

```diff
@@ -0,0 +1,67 @@
+---
+dip: 12
+title: Blockchain Coin Trade Flow
+authors: Lu Zhang (@LuZhang-Lou)
+dip manager: David Wolinsky (@davidiw)
+status: Draft
+type: Informational
+created: 3/5/2021
+---
+# Summary and Motivation
+This document proposes a protocol standard for coin trade flows in the Diem Network. The flow is generic for coin purchase and coin sell transactions involving one or more Designated Dealers (DD). It supports a single trade settlement and batch settlement.
+
+A Blockchain coin trade is the process where an entity, for example a Virtual Asset Service Provider (VASP), transacts with a DD to purchase or sell coins. Beyond on-chain p2p transactions, a coin trade usually involves several stages ranging from trade initialization, fiat currency (or other types of cryptocurrency) debitted, to on-chain transfer. The first few steps may vary from entities to entities. However, we see the possibilities and benefits to provide a general specification for the on-chain phase, in pursuit of interoperability among Diem Blockchain participants.
+
+# Terminology
+* trade_id
+  * a unified trade identifier used over the course, a UTF-8 string, max length 255 bytes. It is agreed upon, by both entities, during the trade initialization and negotiation phase. It should be buyer-seller pair-wise unique.
+
+# Specification
+We propose a new type of on-chain Metadata - `CoinTradeMetadata`.
+
+```
+enum Metadata {
+  Undefined,
+  GeneralMetadata(GeneralMetadata),
+  TravelRuleMetadata(TravelRuleMetadata),
+  UnstructuredBytesMetadata(UnstructuredBytesMetadata),
+  RefundMetadata(RefundMetadat),
+  CoinTradeMetadata(CoinTradeMetadata),
+  /// and other types of metadatas
+}
```

Comment:
should really figure out a more reasonable way of communicating this without copying around this ever increasing blob....

Reply:
We should probably be much more explicit, for example instead of copying this list we should state what the serialized format looks like, e.g. what is the variant number that corresponds to this entry